### PR TITLE
Checkstyle Rules

### PR DESCRIPTION
**Pull request by garyrussell**

### CODE REVIEW

**Comment by artembilan** [comment]
I think such a work should be reflected in the Release Blog Post.
And that may be as a valuable fact to release `4.3.M2` enough soon.
:smile:

File: src/checkstyle/fixRightCurly.gradle [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+							headerFixed = true
+						}
+					}
+
+					if (error && ln == (error.@line as int)) {
+						def message = error.@message
+
+						def index = (error.@column as int) + 1
+						def chars = line.toCharArray()
+						def nextLine = "\n"
```
Comment (artembilan) [inline src/checkstyle/fixRightCurly.gradle]: `System.lineSeparator()` ?

File: src/checkstyle/fixRightCurly.gradle [hunk 2]
```diff
@@ -0,0 +1,65 @@
+task fixRightCurly << {
+	fileTree("${buildDir}/reports/checkstyle").include('*.xml').each { report ->
+		def xml = new XmlParser(false, false).parse(report)
+		xml.file.each { f ->
+			def errors = f.error
+			def thisErrors = []
+			errors.each { error ->
+				if (error.@source == 'com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck') {
+					thisErrors.add(error)
```
Comment (artembilan) [inline src/checkstyle/fixRightCurly.gradle]: Minor, of course, but this isn't `thisErrors`
`rightCurlyErrors`? :smile:

**Comment by artembilan** [comment]
How have you fixed all of those trailing whitespaces?
I don't see any script.
So, what is the magic?

Thanks

**Comment by garyrussell** [comment]
shell command - see https://github.com/spring-projects/spring-framework/wiki/Spring-Framework-Code-Style

**Comment by garyrussell** [comment]
There's a lot more to do, but I need a break from this stuff :smile:

**Comment by garyrussell** [comment]
Spring AMQP is in good shape, though.

File: src/checkstyle/fixRightCurly.gradle (hunk 2 above)
Comment (garyrussell) [inline src/checkstyle/fixRightCurly.gradle]: I think we'll end up removing all these when we're done so not worth too much polish here.

File: src/checkstyle/fixRightCurly.gradle (hunk 1 above)
Comment (garyrussell) [inline src/checkstyle/fixRightCurly.gradle]: :+1:

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration/commit/842aded9a4439b717ba8aac1798dfa2a5680aadf

**Pull request closed without merging**